scripts/gif_maker_parallelized.py

```diff
@@ -0,0 +1,125 @@
+"""
+Script to generate gifs from traj
+
+Requirements:
+
+sudo apt update
+sudo apt install povray
+sudo apt install ffmpeg
+pip install ase==3.20
+
+"""
+import argparse
+import copy
+import multiprocessing as mp
+import os
+
+import ase.io
+import numpy as np
+from ase.data import covalent_radii
+from ase.io.pov import get_bondpairs
+
+
+def pov_from_atoms(mp_args):
+
+    atoms, idx, out_path = mp_args
+    # how many extra repeats to generate on either side to look infinite
+    extra_cells = 2
+    # try and guess which atoms are adsorbates since the tags aren't correct after running in vasp
+    # ideally this would be fixed by getting the right adsorbate atoms from the initial configurations
+    atoms_organic = np.array(
+        [atom.symbol in set(["C", "H", "O", "N"]) for atom in atoms]
+    )
+    # get the bare surface (note: this will not behave correctly for nitrides/hydrides/carbides/etc)
+    atoms_surface = atoms[~atoms_organic].copy()
+    # replicate the bare surface
+    atoms_surface = atoms_surface.repeat(
+        (extra_cells * 2 + 1, extra_cells * 2 + 1, 1)
+    )
+    # make an image of the adsorbate in the center of the slab
+    atoms_adsorbate = atoms[atoms_organic]
+    atoms_adsorbate.positions += extra_cells * (
+        atoms.cell[0, :] + atoms.cell[1, :]
+    )
+    # add the adsorbate to the replicated surface, then center the positions on the adsorbate
+    num_surface_atoms = len(atoms_surface)
+    atoms_surface += atoms_adsorbate
+    atoms_surface.positions -= atoms_adsorbate.positions.mean(axis=0)
+    # only include bonds for the adsorbate atoms
+    bondpairs = get_bondpairs(atoms_surface)
+    bondpairs = [
+        bond
+        for bond in bondpairs
+        if bond[0] >= num_surface_atoms and bond[1] >= num_surface_atoms
+    ]
+    # write the image with povray
+    bbox = (-6.4, -4, 6.4, 4)  # clip to a small region around the adsorbate
+    os.chdir(f"{out_path}")
+    renderer = ase.io.write(
+        "snapshot_%04i.pov" % idx,
+        atoms_surface,
+        povray_settings={
+            "celllinewidth": 0,
+            "canvas_height": 300,
+            "textures": ["intermediate"] * len(atoms_surface),
+            "bondatoms": bondpairs,
+        },
+        bbox=bbox,
+        rotation="-40x",
```

Comment:
Are these flat universal settings? I want to say in the past I had to modify this and `bbox` for weird angled surfaces? Maybe argeparse these or just make a note in the header

Reply:
There are just too many settings here that can be tuned (including that of povray and ffmpeg), it would be too much to argparse all of them? Maybe I'll just add a note as you mentioned.

Reply:
Yup agreed, documentation should suffice so people know at least what to change.